civihr_employee_portal/views/civihr_employee_portal.views.inc

```diff
@@ -1080,20 +1083,20 @@ function removeInactiveRoleDetails($results = []) {
  * @param views_plugin_query_default $query
  */
 function civihr_employee_portal_views_query_alter(&$view, &$query) {
-  if ($view->name !== 'civihr_staff_directory') {
-    return;
-  }
 
-  $queryParts = [];
-  getWhereFields($query->where, $queryParts);
-  $departmentField = 'hrjc_role_hrjc_revision.role_department';
-  $roleEndField = 'hrjc_role_hrjc_revision.role_end_date';
+  switch ($view->name) {
+    case StaffDirectoryView::getName():
+      $viewClass = new StaffDirectoryView();
+      break;
+    case MyDetailsView::getName():
+      $viewClass = new MyDetailsView();
+      break;
+    default:
+      $viewClass = NULL;
+  }
 
-  // if department is set we must limit to only current roles
-  if (!empty($queryParts[$departmentField])) {
-    $group = $query->set_where_group('OR'); // add a new WHERE group
-    $query->add_where($group, $roleEndField, date('Y-m-d'), '>=');
-    $query->add_where($group, $roleEndField, NULL, 'IS');
+  if ($viewClass) {
```

Comment:
this if can be avoided if return instead of setting `$viewClass` to NULL in default

Reply:
Nice